hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -1,24 +1,56 @@
 package ca.bc.gov.hlth.hnsecure;
 
-
+import java.nio.charset.Charset;
+import java.util.Base64;
 import java.util.Properties;
 
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.PropertyInject;
 import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.http.HttpComponent;
+import org.apache.camel.spi.Registry;
+import org.apache.camel.support.jsse.KeyManagersParameters;
+import org.apache.camel.support.jsse.KeyStoreParameters;
+import org.apache.camel.support.jsse.SSLContextParameters;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
 
 import ca.bc.gov.hlth.hnsecure.authorization.ValidateAccessToken;
 import ca.bc.gov.hlth.hnsecure.exception.CustomHNSException;
+import ca.bc.gov.hlth.hnsecure.json.Base64Encoder;
+import ca.bc.gov.hlth.hnsecure.json.pharmanet.ProcessV2ToPharmaNetJson;
 import ca.bc.gov.hlth.hnsecure.message.ValidationFailedException;
 import ca.bc.gov.hlth.hnsecure.messagevalidation.ExceptionHandler;
 import ca.bc.gov.hlth.hnsecure.messagevalidation.V2PayloadValidator;
 import ca.bc.gov.hlth.hnsecure.parsing.FhirPayloadExtractor;
+import ca.bc.gov.hlth.hnsecure.parsing.PharmaNetPayloadExtractor;
 import ca.bc.gov.hlth.hnsecure.parsing.PopulateReqHeader;
+import ca.bc.gov.hlth.hnsecure.parsing.Util;
 import ca.bc.gov.hlth.hnsecure.properties.ApplicationProperties;
 import ca.bc.gov.hlth.hnsecure.temporary.samplemessages.SampleMessages;
 
 public class Route extends RouteBuilder {
 
+    private static final String KEY_STORE_TYPE_PKCS12 = "PKCS12";
+    
+    // PharmaNet Endpoint values
+	@PropertyInject(value = "pharmanet.uri")
+    private String pharmanetUri;
+
+	@PropertyInject(value = "pharmanet.cert")
+    private String pharmanetCert;
+    
+    private static final String pharmanetCertPassword = System.getenv("PHARMANET_CERT_PASSWORD");
+    
+    private static final String pharmanetUser = System.getenv("PHARMANET_USER");
+
+    private static final String pharmanetPassword = System.getenv("PHARMANET_PASSWORD");
+    
+    public Route() {
+    }
+
     @Override
-    public void configure() {
+    public void configure() throws Exception {
```

Comment:
I believe we should handle all exceptions. 
Scenario 1: If exception in the configuration.. hnsecure server should log that exception and should not start. I have add this in other branch (HNsecure_ResolveTODO). That code will be merged after code review of that branch, so this will be taken care.
Scenario 2: If exception during processing of request, we should handle that in our custom exception handler (discussed with David)

Reply:
This exception was not needed so has been removed.